jenkins/jenkinsfiles/build-charm-k8s.jenkinsfile

```diff
@@ -0,0 +1,114 @@
+                script {
+                    checkout changelog: false, poll: false, scm: [
+                        $class: 'GitSCM',
+                        branches: [[name: "*/test"]],
+                        extensions: [
+                            [$class: 'WipeWorkspace'],
+                        ],
+                        userRemoteConfigs: [
+                            [
+                                credentialsId: "${githubCredId}",
+                                url: 'https://github.com/canonical/candid'
+                            ]
+                        ]
+                    ]
+                    // TODO: This is a workaround to make sure the scripts always exist.
+                    // Eventually once this is merged to develop we need to change target branch to develop
+                    // and then we will always have them.
+                    //
+                    // And once we have a jenkins 2.7+ we can use a dynamically loaded shared
+                    // library, but until then this is what we're working with.
+                    sh 'git checkout origin/test ./jenkins/jenkinsfiles/utils'
+                }
+            }
+        }
+        stage('Setup environment') {
+            steps {
+                script {
+                    lxc = load("${env.WORKSPACE}/jenkins/jenkinsfiles/utils/lxc.groovy")
+                    echo "utils loaded"
+                    lxc.launchContainer('ubuntu:20.04')
+                    echo "container launched"
+                    lxc.pushWorkspace()
+                    echo "workspace pushed"
+                    lxc.s('pwd')
+                    lxc.installSnap('go', true, '1.18/stable')
+                    lxc.installSnap('charmcraft', true, 'latest/stable')
+                    lxc.s('sudo apt update')
+                    lxc.s('sudo apt-get update')
+                    lxc.s('sudo apt install make')
+                    lxc.s('sudo apt-get install build-essential autoconf python-dev-is-python3 -y') // GCC for CGO
+                    usr = "${env.GITHUB_PAT_AUTH_USR}"
+                    psw = "${env.GITHUB_PAT_AUTH_PSW}"
+                    setGitCreds = """
+                    git config \
+                    --global \
+                    url."https://${usr}:${psw}@github.com".insteadOf \
+                    "https://github.com"
+                    """
+                    lxc.s(setGitCreds)
+                }
+                echo 'Environment setup complete.'
+            }
+        }
+        stage('Build charm') {
+            steps {
+                script {
+                    lxc.s('sudo charmcraft pack --verbose --destructive-mode --project-dir ./charms/candid-k8s')
+                }
+            }
+        }
+        stage('Archive charm') {
+            steps {
+                script {
+                    String getCharmNamCmd = """echo "\$(cat ./charms/candid-k8s/metadata.yaml | yq .name)_ubuntu-20.04-amd64.charm" """
```

Comment:
The charm name as defined here could change if the "run-on" parameters in charmcraft.yaml are changed.